src/include/ipc/topology.h

```diff
@@ -87,6 +87,8 @@ struct sof_ipc_comp {
 #define SOF_MEM_CAPS_EXEC		BIT(7) /**< executable */
 #define SOF_MEM_CAPS_L3			BIT(8) /**< L3 memory */
 
+#define SOF_MEM_CAPS_MAX_BIT		BIT(8) /**< Used for input validation */
```

Comment:
Since this is ABI, its better we make it obvious via an inline comment that MAX_BIT needs to be updated for any new MEM_CAPS addition. I would also define MAX_BIT as CAPS_L3 rather than BIT(8) as this makes it even clearer to anyone reading the code.

Reply:
> we make it obvious via an inline comment that MAX_BIT needs to be updated for any new MEM_CAPS addition.

Comment added.

> I would also define MAX_BIT as CAPS_L3 rather than BIT(8) as this makes it even clearer to anyone reading the code.

I can't do that because I realized v2 12aff94afee06c46274628503ee337c801ff6314 was buggy, apologies. In v2 I made _MAX equal to the last last valid bit but that is wrong, it would incorrectly reject _MAX + some other BIT. The tests passed anyway because we don't seem to test L3_HEAP.

The comparison must be with the first invalid bit. I confused myself with the name "_MAX", it is a bad name. I just replaced it with _LOWEST_INVALID.